packages/react/src/hooks/accounts/useSigner.ts

```diff
@@ -28,28 +32,22 @@ export function useSigner<TSigner extends Signer>({
   onSettled,
   onSuccess,
 }: UseSignerConfig = {}) {
+  const { connector } = useAccount()
   const chainId = useChainId({ chainId: chainId_ })
   const signerQuery = useQuery<
     FetchSignerResult<TSigner>,
     Error,
     FetchSignerResult<TSigner>,
     ReturnType<typeof queryKey>
-  >(queryKey({ chainId }), queryFn, {
+  >(queryKey({ chainId, connectorId: connector?.id }), queryFn, {
```

Comment:
Will `connectorId` be unique enough for when people use multiple Connectors (e.g. connect modal libraries)? We only allow one "active" Connector right now in the wagmi Client, but maybe in the future we can allow multiple - does that break this assumption?

Reply:
True. Might be best if we can make this more unique. Will fix!